### PR TITLE
Make columns indices 1-based in the text output format

### LINKED ISSUE #537: Report errors using 1-indexed columns
See: #414

### PR DESCRIPTION
**Pull request by fsouza**
Not changing the column in JSON errors. Not sure if we want to change that too.

Closes #537.

### CODE REVIEW

**Comment by charliermarsh** [comment]
I think it'd make sense to change it in the JSON printer too if you're up for it. Should we even consider changing it when we convert from `Check` to `Message`, so that `Message` is always one-indexed?

**Comment by fsouza** [comment]
@charliermarsh good call! Let me make that change.

**Comment by fsouza** [comment]
@charliermarsh how does this look? I originally tried to make a copy of the `Location` directly, but the fields are private, so I ended-up copying it and then mutating with `go_right()`.

My Rust-fu is a bit limited here, let me know if I'm wasting your time 😁

File: src/message.rs [hunk 1]
```diff
@@ -18,6 +18,22 @@ pub struct Message {
     pub filename: String,
 }
 
+impl Message {
+    pub fn from_check(filename: String, check: Check) -> Self {
+        let mut location = check.location;
+        location.go_right();
+        let mut end_location = check.end_location;
+        end_location.go_right();
```
Comment (fsouza) [inline src/message.rs:26]: this doesn't feel right 🙈

Comment (charliermarsh) [reply src/message.rs:26]: I think you can use Location::new and location.row() methods to create a new location here! (On subway so low-fidelity message…)

Comment (fsouza) [reply src/message.rs]: Ohh that makes more sense hah I was too influenced by Go in my first attempt. Pushed a fix!

Comment (charliermarsh) [reply src/message.rs]: I think you can use Location::new and location.row() methods to create a new location here! (On subway so low-fidelity message…)

Comment (charliermarsh) [reply src/message.rs]: Woo, thanks!

**Comment by charliermarsh** [comment]
\cc @Seamooo - we changed `Message` back to one-based indexing for columns. (`Check` still uses zero-based indexing, so if `ruffd` uses `Check` now as planned, it will be unaffected.)

**Comment by Seamooo** [comment]
Awesome, ruffd also now uses location and end_location in `Patch` and this also appears unaffected

**Pull request merged**